internal/scheduler/ingester.go

```diff
@@ -0,0 +1,777 @@
+				JobSet:         sequence.GetJobSetName(),
+				UserID:         sequence.GetUserId(),
+				Groups:         sequence.GetGroups(),
+				Queue:          sequence.GetQueue(),
+				Priority:       int64(e.SubmitJob.Priority),
+				SubmitMessage:  submitJobBytes,
+				SchedulingInfo: schedulingInfoBytes,
+			})
+		case *armadaevents.EventSequence_Event_JobRunLeased:
+			batch.Runs = append(batch.Runs, Run{
+				RunID:    armadaevents.UuidFromProtoUuid(e.JobRunLeased.GetRunId()),
+				JobID:    armadaevents.UuidFromProtoUuid(e.JobRunLeased.GetJobId()),
+				JobSet:   sequence.GetJobSetName(),
+				Executor: e.JobRunLeased.GetExecutorId(),
+			})
+		case *armadaevents.EventSequence_Event_ReprioritiseJob:
+			if len(batch.Reprioritisations) == 0 {
+				batch.Reprioritisations = append(batch.Reprioritisations, NewReprioritisationBatch())
+			}
+			reprioritisation := batch.Reprioritisations[len(batch.Reprioritisations)-1]
+			newPriority := int64(e.ReprioritiseJob.GetPriority())
+			if priority, ok := reprioritisation.PrioritiesByJobSet[sequence.JobSetName]; ok && priority == newPriority {
+				break // This operation is redundant.
+			}
+			jobId := armadaevents.UuidFromProtoUuid(e.ReprioritiseJob.GetJobId())
+			reprioritisation.PrioritiesByJob[jobId] = newPriority
+		case *armadaevents.EventSequence_Event_ReprioritiseJobSet:
+			if len(batch.Reprioritisations) == 0 {
+				batch.Reprioritisations = append(batch.Reprioritisations, NewReprioritisationBatch())
+			}
+			reprioritisation := batch.Reprioritisations[len(batch.Reprioritisations)-1]
+
+			// To ensure the most priority last written to Pulsar is applied last,
+			// and since we apply ReprioritiseJobSet before ReprioritiseJob messages for each ReprioritisationBatch,
+			// we need to create a new ReprioritisationBatch if len(reprioritisation.PrioritiesByJob) != 0.
+			if len(reprioritisation.PrioritiesByJob) != 0 {
+				batch.Reprioritisations = append(batch.Reprioritisations, NewReprioritisationBatch())
+				reprioritisation = batch.Reprioritisations[len(batch.Reprioritisations)-1]
+			}
+
+			newPriority := int64(e.ReprioritiseJobSet.GetPriority())
+			reprioritisation.PrioritiesByJobSet[sequence.GetJobSetName()] = newPriority
+		case *armadaevents.EventSequence_Event_CancelJobSet:
+			batch.JobSetsCancelled[sequence.GetJobSetName()] = true
+		case *armadaevents.EventSequence_Event_CancelJob:
+			jobId := armadaevents.UuidFromProtoUuid(e.CancelJob.GetJobId())
+			batch.JobsCancelled[jobId] = true
+		case *armadaevents.EventSequence_Event_JobSucceeded:
+			jobId := armadaevents.UuidFromProtoUuid(e.JobSucceeded.GetJobId())
+			batch.JobsSucceeded[jobId] = true
+		case *armadaevents.EventSequence_Event_JobErrors:
+			eventId := eventid.New(msg.ID(), i).String()
+			for j, jobError := range e.JobErrors.GetErrors() {
+				bytes, err := proto.Marshal(jobError)
+				if err != nil {
+					err = errors.WithStack(err)
+					logging.WithStacktrace(log, err).Error("failed to marshal JobError")
```

Comment:
I'd be tempted to set set `bytes` to nil here on the grounds that if a function returns an error I wouldn't use the value.